.github/workflows/build-to-cloudflare-pages.yml

```diff
@@ -0,0 +1,124 @@
+    cancel-in-progress: true
+
+jobs:
+    build_to_cloudflare_pages:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout to repo
+              uses: actions/checkout@v3
+              with:
+                  ref: ${{ github.event.pull_request.head.sha }}
+
+            - name: Setup node
+              uses: actions/setup-node@v2
+
+            - name: Generate build status comment
+              id: build_status_comment
+              uses: actions/github-script@v3
+              with:
+                  script: |
+                      const preview_url = "https://" + "${{ github.head_ref }}" + ".deriv-com.pages.dev"
+                      const comment = [
+                        '## Cloudflare Pages Preview link',
+                        '| Name | Result |',
+                        '| :--- | :------ |',
+                        `| **Build status**  | In progress ⏳ |`,
+                        `| **Preview URL**  | [Visit Preview](${preview_url}) |`,
+                        ''
+                      ].join('\n')
+                      core.setOutput("comment", comment);
+
+            - name: Update build status comment
+              uses: marocchino/sticky-pull-request-comment@v2
+              with:
+                  only_update: true
+                  header: Cloudflare Pages Preview Comment
+                  number: ${{github.event.issue.number}}
+                  message: ${{steps.build_status_comment.outputs.comment}}
+
+            - name: Get build output from master cache
+              uses: actions/cache/restore@v3
+              with:
+                  key: master-cache-public
+                  restore-keys: |
+                      master-cache-public-replica
+                  path: |
+                      .cache
+                      public
+
+            - name: Get cached dependencies
+              id: cache-npm
+              uses: actions/cache/restore@v3
+              with:
+                  path: node_modules
+                  key: npm-${{ hashFiles('**/package-lock.json') }}
+
+            - name: Install dependencies
+              if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+              run: npm ci
+
+            - name: Build project
+              id: build-project
+              run: npm run build
+
+            - name: Publish to Cloudflare Pages
+              id: publish-to-pages
+              run: |
+                  echo "Installing Wrangler CLI"
+                  npm i -g wrangler@3.5.0
+
+                  echo "Deploying build to Cloudflare Pages"
+                  directory='public'
+                  projectName='deriv-com'
+                  branch=$GITHUB_HEAD_REF
+
+                  preview_url=$(wrangler pages deploy $directory --project-name=$projectName --branch=$branch > log.txt 2>&1; echo $?)
+                  echo "------"
+                  cat log.txt
+
+                  branchName=$(echo $branch | sed 's/[\/_]/-/g')
```

Comment:
Cloudflare Pages formats the default preview link to use `-`, so if a branch has a name like `feq-123/bla-bla`, the preview link will show it as `https://feq-123-bla-bla.deriv-com-playground.pages.dev`

Reply:
<img width="577" alt="Screenshot 2023-08-10 at 5 54 08 PM" src="https://github.com/binary-com/deriv-com/assets/103016120/5d5ae8d0-cd19-4716-b936-7a7b27d92dad">